### DISCUSSION TITLE
How do I properly format quarto render to match markdown render syntax?

### DISCUSSION THREAD

**Discussion by NagyO-MQ** [quarto-dev/quarto-cli#7524] [Q&A] [answered]
### Description

I've been using a purr loop in R (based on this blog: https://damien-datasci-blog.netlify.app/post/2022-01-06-generating-multiple-pdf-at-once-a-use-case-of-rmarkdown-render-in-a-purrr-loop/):

purrr::walk(
  .x = DataSet$Column,
  ~ rmarkdown::render(
    input = "MarkdownTemplate.rmd",
    output_file = glue::glue("Title{.x}.Rmd"),
    params = list(Column = {.x})))

It generates a set of multiple markdown documents at once based on a markdown template where the data for each is filled according to a subset of the dataset (1 of the columns), which is set as a parameter. 

I'm trying to adapt this to quarto but can't work out the syntax for quarto_render. Does anyone have some advice on this?

Thanks!

**Comment by cderv** [comment]
It should work quite similarly unless there is an issue with Quarto R package (https://github.com/quarto-dev/quarto-r) 

Only thing to whatch out for is that `params` in Quarto render is a list thatt will be converted (serialized) to JSON so that i can be passed to Quarto (https://quarto.org/docs/computations/parameters.html). You can pass complex R object like data.frame. For this you need to go through a `.Rds` file for example, or pass a variable name to subset in the doc. 

Also in your example you are rendering from a .Rmd to a .Rmd, so I am not sure which format your are using in **rmarkdown**. A full reproducible example would help. 

> It generates a set of multiple markdown documents at once based on a markdown template where the data for each is filled according to a subset of the dataset (1 of the columns), which is set as a parameter.

For this, you may want to consider simple template tool that do not require a render at all. See **brew** or **whisker** or even `knitr::knit_expand()` or even `glue::glue()` by reading and writing back. 

Depending on what you want to do, `knit_expand()` + `knit_child()` could be helpful too. See this example about Tabsets creation using R chunk: https://examples.quarto.pub/create-tabsets-panel-from-r-code/ and the links to R Markdown Cookbook at the bottom. 

Hope this helps

**Reply by NagyO-MQ** [reply]
Thanks for your response! The code I use to generate the files is modified to read qmd files as follows:

```
     library(tidyverse)
     library(tinytex)
     library(bookdown)
     library(openxlsx)
     library(knitr)
     library(quarto)

     scalelist = read.xlsx("AbridgedScaleSummary.xlsx")

     purrr::walk(
        .x = scalelist$abbreviation,
        ~ quarto_render(
        input = "QuartoProjTemplate.qmd",
        output_file = glue::glue("Scale--{.x}.qmd"),
        execute_params = list(abbreviation = {.x})))
```
    
The template is a qmd file (QuartoProjTemplate) with the following yaml:

---
title: "ASC Scale - `r params$abbreviation`"
params:
  abbreviation: "Scale"
---

and the following parameter set:
```
     scalelist <- 
       read.xlsx("AbridgedScaleSummary.xlsx") |>
       filter(abbreviation==params$abbreviation)
```

At present I just get the following error:

Error in `map()`:
ℹ In index: 1.
Caused by error:
! System command 'quarto' failed
---
Backtrace:
  1. purrr::walk(...)
  2. purrr::map(.x, .f, ..., .progress = .progress)
  3. purrr:::map_("list", .x, .f, ..., .progress = .progress)
  6. global .f(.x[[i]], ...)
  7. quarto::quarto_render(...)
  8. processx::run(quarto_bin, args, echo = TRUE)
  9. throw(...)
 10. base::signalCondition(cond)

I'll look into those alternatives as well if you can't see an obvious way I'm going wrong here.

**Reply by mcanouil** [reply]
Your example is not reproducible. The error you encounter can be at any level including your R code.
Could you properly format your post for it to be legible, especially the code such as the YAML?

**Reply by cderv** [reply]
@NagyO-MQ please see https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us to help you format correctly. Thanks

**Reply by NagyO-MQ** [reply, marked as answer]
Apologies, I'm new to asking questions here. I did manage to resolve the issue. The .yml file has to be created after the qmd files have been rendered, otherwise it tries to use the .yml to format them, which causes the error. Thank you anyway for your time!
